src/Core/Grid/Definition/Factory/CustomerAddressGridDefinitionFactory.php

```diff
@@ -54,7 +54,22 @@ final class CustomerAddressGridDefinitionFactory extends AbstractGridDefinitionF
     public function __construct(HookDispatcherInterface $hookDispatcher, ?Request $currentRequest)
     {
         parent::__construct($hookDispatcher);
-        $this->backUrl = $currentRequest ? $currentRequest->getUri() : '';
+
+        // Remove query parameters to ensure behavior of "customer address grid" is consistent with `AddressGridDefinitionFactory`
+        if ($currentRequest) {
+            $requestCopy = clone $currentRequest;
+
+            $query = $requestCopy->query;
+            $query->remove('customer_address');
```

Comment:
I don't understand why the customer condition is removed?🤔
From what I understand in the issue, this customer address grid is destined to display only the list of address of a designated customer no? So it means after removing a customer address we redirect to the generic address URL?

I don't think it's the goal here. The problem is about showing an empty page after removal, which means we don't perform any check on page validity in our grid system, but it is likely that this is a more global bug that affects all the grids so it should be fixed globally in the grid system IMHO

Reply:
I tested other grids seems like the user is redirected to the first page because we simply use the list route usually
I can't test on develop because of another bug, but I guess this ugly use of currentRequest is here because the customerId is present in the URL

So the idea to fix this problem would be to find a way to extract the customerId from the URL so that we can rebuild the `admin_customers_view` with the appropriate parameters

If we can find a way to do this without relying on the Request injection it would be great, but I'm not sure exactly why and the idea is not to refacto everything ^^

Reply:
If I remember well, abstract grid redirect to first page after a delete but this page use a "custom" field cause there are several grids. This field allows to navigate in a given grid without changing others parameters.

If you want some refactoring of grid system I am open to suggestions but i would enjoy some deeper introduction to the component, the global architecture direction, etc.

Reply:
So if I get it right the purpose is to remove all the filters parameters for the `CustomerAddressGrid` without removing the ones from the `CustomerDiscountGrid` right? But how comes the discount grid doesn't do the same then? 🤔

I can't help but think this was poorly implemented from the start and we're only trying to put a bandage on something that is half broken already. It's not against your modification but it gives me chills whenever the request is forcefully injected in a service like this, I've had many troubles in the past when this was done and I can't help but think it's a very bad practice.

The discount deletion is not handled the same way, it created a dedicated action:
https://github.com/PrestaShop/PrestaShop/blob/f332ae8a1c6f75e83a620c3b96bf4c3060fcbc6f/src/Core/Grid/Action/Row/Type/Customer/DeleteCustomerDiscountRowAction.php#L39

with a dedicated template in which the back attribute is used, but it's less shocking to do this here for me:
https://github.com/PrestaShop/PrestaShop/blob/83977178c98e95be1ca6b7393d16a40766feb8c5/src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Actions/Row/delete_customer_discount.html.twig#L37

Reply:
That said since the discount filter's parameters are not cleaned from the URI I guess there will be the same bug on the other grid? Did you test it?

Reply:
IMHO it would be cleaner to use `admin_customers_view` as the back URL it will delete both filters for address grid and discount grid but do we really need to keep the pagination for both grids after a deletion? Besides the filters are supposed to be persisted in the database so they should be still used on refresh even if they are not present in the URL theoretically. I'm even surprised that your solution to simply remove them from the URI works as the DB filters should be used 🤔

Reply:
Sorry, I missed your comments and it is so old now I need time to retest and understand what I did and why xD

I am not sure that I understood well what you want.

Do you want `PrestaShop\PrestaShop\Core\Grid\Definition\Factory\CustomerAddressGridDefinitionFactory` and `PrestaShop\PrestaShop\Core\Grid\Definition\Factory\CustomerDiscountGridDefinitionFactory` behave the same way (which would imply some refactoring) ?

Do you want me to edit `src/PrestaShopBundle/Resources/config/services/core/grid/grid_definition_factory.yml` and `CustomerAddressGridDefinitionFactory::__construct` to use directly `admin_customers_view` as back url instead of using `Request` directly ?
(This would create a Breaking Change)